source/manual/how-to-upgrade-rails.html.md

```diff
@@ -0,0 +1,194 @@
+   upgrades. Documenting this will make decisions easier when performing future
+   upgrades.
+
+[GDS guidance on commit messages]:
+  https://gds-way.digital.cabinet-office.gov.uk/standards/source-code/working-with-git.html#commit-messages
+
+#### Guidance for all upgrades
+
+The following guidance should be considered secondary to the [version-specific
+guidance][] that follows.
+
+- Accept changes to comments that come from Rails.
+- `config/puma.rb` will often be set up to use [`GovukPuma`][govuk-puma] rather
+  than Rails defaults. Ignore changes in this case.
+- For changes to settings in `config/environment`, work out if we were using a
+  previous Rails default.
+
+  If we weren't using a previous Rails default, look for an explanation in the
+  app's git history.
+
+  If we were using a previous Rails default, consult the following to inform
+  your decision on whether to move to the new default:
+
+  - [GOV.UK Helm Charts][] - does the config change affect an environment
+    variable that we've set for the app?
+  - The app’s git history
+  - [Rails version-specific guidance][official Rails guidance]
+  - [Rails source code][] and git history related to the change
+  - [Rails change log][] for the given release
+  - [Rails configuration documentation][]
+
+- For changes in `public/*`, in general you can leave out HTML and image files
+  (and consider removing any existing ones). We serve our own error pages and
+  don't use Rails default app icons.
+- It's useful to commit the `config/new_framework_defaults_*.rb` file. These are
+  new settings that will be used when you [update the version defaults
+  setting][]. With this file committed, if there are any issues after switching
+  to the new defaults, you can try reverting to an earlier version's defaults
+  and turning on these settings one by one.
+
+If you're still unsure about a change:
+
+- document your uncertainty in the commit message;
+- flag the change when opening a pull request;
+- ask your team or other Ruby colleagues for help.
+
+[govuk-puma]:
+  https://github.com/alphagov/publishing-api/blob/9b38ea8df732273fb99244a9366f41096474e714/config/puma.rb
+[GOV.UK Helm Charts]: https://github.com/alphagov/govuk-helm-charts
+[Rails change log]: https://github.com/rails/rails/releases
+[Rails configuration documentation]:
+  https://guides.rubyonrails.org/configuring.html
+[Rails source code]: https://github.com/rails/rails
+[update the version defaults setting]: #switch-to-the-new-version-defaults
+[version-specific guidance]: #guidance-for-upgrades-to-specific-versions
+
+#### Guidance for upgrades to specific versions
+
+Version-specific guidance is available, starting from Rails 8.0:
+
+- [8.0][]
+
+[8.0]: /manual/guidance-for-upgrades-to-rails-8-0
+
+### Switch to the new version defaults
+
+1. Update the version number provided to the `config.load_defaults` setting in
+   `config/application.rb`.
+1. Delete the `config/new_framework_defaults_*.rb` file.
```

Comment:
I think this file exists at `config/initializers/new_framework_defaults_*.rb`, rather than at the top level of the `config` directory.

Reply:
Oops, thanks for spotting!